plugins/in_winevtlog/winevtlog.c

```diff
@@ -0,0 +1,786 @@
+            wEventXML = (LPWSTR)flb_malloc(dwBufferSize);
+            if (wEventXML)
+            {
+                EvtRender(NULL, hEvent, flags, dwBufferSize, wEventXML, &dwBufferUsed, &count);
+            }
+            else
+            {
+                flb_error("malloc failed");
+                goto cleanup;
+            }
+        }
+
+        if (ERROR_SUCCESS != (status = GetLastError()))
+        {
+            flb_error("EvtRender failed with %d", GetLastError());
+            goto cleanup;
+        }
+    }
+
+    return wEventXML;
+
+cleanup:
+
+    if (wEventXML)
+        flb_free(wEventXML);
+
+    return NULL;
+}
+
+DWORD render_system_event(EVT_HANDLE hEvent, PEVT_VARIANT *pSystem, unsigned int *system_size)
+{
+    DWORD status = ERROR_SUCCESS;
+    EVT_HANDLE hContext = NULL;
+    DWORD dwBufferSize = 0;
+    DWORD dwBufferUsed = 0;
+    DWORD dwPropertyCount = 0;
+    PEVT_VARIANT pValues = NULL;
+
+    hContext = EvtCreateRenderContext(0, NULL, EvtRenderContextSystem);
+    if (NULL == hContext) {
+        status = GetLastError();
+        flb_error("failed to create RenderContext with %d", status);
+
+        goto cleanup;
+    }
+    if (!EvtRender(hContext,
+                   hEvent,
+                   EvtRenderEventValues,
+                   dwBufferSize,
+                   pValues,
+                   &dwBufferUsed,
+                   &dwPropertyCount)) {
+        status = GetLastError();
+
+        if (ERROR_INSUFFICIENT_BUFFER == status) {
+            dwBufferSize = dwBufferUsed;
+            pValues = (PEVT_VARIANT)flb_malloc(dwBufferSize);
+            if (pValues) {
+                EvtRender(hContext,
+                          hEvent,
+                          EvtRenderEventValues,
+                          dwBufferSize,
+                          pValues,
+                          &dwBufferUsed,
+                          &dwPropertyCount);
+                status = GetLastError();
+                *system_size = dwBufferUsed;
+            } else {
+                if (pValues)
+                    flb_free(pValues);
+
+                flb_error("failed to malloc memory with %d", status);
+
+                goto cleanup;
+            }
+        }
+
+        if (ERROR_SUCCESS != status) {
+            EvtClose(hContext);
+            flb_free(pValues);
+
+            return status;
+        }
+    }
+
+    *pSystem = pValues;
+
+cleanup:
+
+    if (hContext)
+        EvtClose(hContext);
+
+    return status;
+}
+
+
+PWSTR get_message(EVT_HANDLE hMetadata, EVT_HANDLE handle, unsigned int *message_size)
+{
+    WCHAR* pBuffer = NULL;
+    DWORD status = ERROR_SUCCESS;
+    DWORD dwBufferSize = 0;
+    DWORD dwBufferUsed = 0;
+    LPVOID lpMsgBuf;
+    WCHAR* wMessage = NULL;
+    char *error_message = NULL;
+
+    // Get the size of the buffer
+    if (!EvtFormatMessage(hMetadata, handle, 0, 0, NULL, EvtFormatMessageEvent, dwBufferSize, pBuffer, &dwBufferUsed)) {
```

Comment:
You might want to break this line up

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/4179/commits/a6583c0c04c813dc128074f8223dd90a3777ce78